src/java.base/share/classes/sun/security/ssl/NewSessionTicket.java

```diff
@@ -375,43 +362,104 @@ public byte[] produce(ConnectionContext context) throws IOException {
             // get a new session ID
             SSLSessionContextImpl sessionCache = (SSLSessionContextImpl)
                 hc.sslContext.engineGetServerSessionContext();
-            SessionId newId = new SessionId(true,
-                hc.sslContext.getSecureRandom());
+            int sessionTimeoutSeconds = sessionCache.getSessionTimeout();
+            if (sessionTimeoutSeconds > MAX_TICKET_LIFETIME) {
+                if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
+                    SSLLogger.fine("No session ticket produced: " +
+                            "session timeout");
+                }
+
+                return null;
+            }
 
             SecretKey resumptionMasterSecret =
                 hc.handshakeSession.getResumptionMasterSecret();
             if (resumptionMasterSecret == null) {
                 if (SSLLogger.isOn && SSLLogger.isOn("ssl,handshake")) {
-                    SSLLogger.fine(
-                            "No session ticket produced: " +
-                            "no resumption secret");
+                    SSLLogger.fine("No session ticket produced: " +
+                        "no resumption secret");
                 }
 
                 return null;
             }
 
-            // construct the PSK and handshake message
-            BigInteger nonce = hc.handshakeSession.incrTicketNonceCounter();
-            byte[] nonceArr = nonce.toByteArray();
-            SecretKey psk = derivePreSharedKey(
-                    hc.negotiatedCipherSuite.hashAlg,
-                    resumptionMasterSecret, nonceArr);
+            /*
+             * This thread addresses a Windows only networking issue found with
+             * SSLSocketBruteForceClose. A client that quickly closes after
+             * TLS Finished completed would cause the server read-side to
+             * get a SocketException: "An established connection was aborted
+             * by the software in your host machine", which relates to WinSock
+             * error WSAECONNABORTED.  This situation was observed with
+             * multiple NST messages when the client and server threads on the
+             * same machine.  This is very unlikely to be seen where client
+             * and server are on different machines.
+             */
+            Thread nstThread = Thread.ofVirtual().name("NST").start(() -> {
```

Comment:
Please don't use threads during handshake.

Reply:
There is no alternative that I have found for this synchronization/timing situation.  We certainly don't want a `sleep()` call and NSTs are not send/ack situation.  If the client ignores the NST, that is fine.  Hung thread paranoia is the only reason I put the `join()` in the code below as when this finish isn't critical.